### PR TITLE
fix: update chat in close to real time

### LINKED ISSUE #850: Group chats are not updated in "real time"
**Expected behaviour**
Messages in group chats should be updated in (near) real time.

**Actual behaviour**
Messages in group chats are currently not refreshed in (near) real time unless:
- The user quits the chat and go back in, or
- The user sends a message

**Other notes**
Should be fixable for now by changing the `refetchInterval` option in the `[groupChatMessages, groupChatId]` query to refetch every 2-3 second (hence "real time" in quotes). Probably should look into how grpc streams might help with this in the longer term as it'll likely be more efficient than spamming the server with this regular polling.

### PR DESCRIPTION
**Pull request by darrenvong**
Closes #850

Still a bit concerned that this might overload the backend when there are enough users even though this is set to refetch every 3 seconds (so it's not quite real time)... hopefully streaming or being able to use websocket so the connection is established once might help with this

### CODE REVIEW

File: app/frontend/src/features/messages/groupchats/GroupChatView.test.tsx [hunk 1]
```diff
@@ -125,6 +126,79 @@ describe("GroupChatView", () => {
...
+    expect(getGroupChatMock).toHaveBeenCalledTimes(1);
+    // 1 is groupChatId here
+    expect(getGroupChatMock).toHaveBeenCalledWith(1);
+    expect(getGroupChatMessagesMock).toHaveBeenCalledTimes(1);
+    expect(getGroupChatMessagesMock).toHaveBeenCalledWith(1, undefined);
+  });
+
+  describe("when there are new messages due to come in", () => {
+    beforeEach(() => {
+      jest.useFakeTimers();
```
Comment (darrenvong) [inline app/frontend/src/features/messages/groupchats/GroupChatView.test.tsx:138]: Lol figuring this out was harder than making the change itself. But glad we have done it so we have an example and also so we don't have to wait for 3 seconds for this test to finish!

File: app/frontend/src/features/messages/groupchats/GroupChatView.tsx [hunk 2]
```diff
@@ -138,8 +145,8 @@ export default function GroupChatView() {
     (text) => service.conversations.sendMessage(groupChatId, text),
     {
       onSuccess: () => {
-        queryClient.invalidateQueries(["groupChatMessages", groupChatId]);
-        queryClient.invalidateQueries(["groupChats"]);
+        queryClient.invalidateQueries(groupChatMessagesKey(groupChatId));
+        queryClient.invalidateQueries([groupChatBaseKey]);
```
Comment (lucaslcode) [inline app/frontend/src/features/messages/groupchats/GroupChatView.tsx]: I don't think this should be named like this. Maybe `groupChatsList`?

**Review by lucaslcode — CHANGES_REQUESTED** [review]

File: app/frontend/src/features/messages/groupchats/GroupChatView.tsx (hunk 2 above)
Comment (darrenvong) [reply app/frontend/src/features/messages/groupchats/GroupChatView.tsx]: Yeah that's a better name, missed another place where this is used so I'll update there too

**Comment by darrenvong** [comment]
Not sure why the build failed in the other steps, but it's not related to the frontend... probably related to the Docker Hub issue Aapeli mentioned earlier

**Review by lucaslcode — APPROVED** [review]

**Pull request merged**